packages/askui-nodejs/src/core/inference-response/inference-response.ts

```diff
@@ -1,41 +1,34 @@
 import { ControlCommand } from '../ui-control-commands/control-command';
 import { Annotation } from '../annotation/annotation';
+import { ModelType } from './model-type';
+import { InvalidModelTypeError } from './invalid-model-type-error';
 
+export interface InferenceResponseBody {
+  type: ModelType;
+  data: ModelType extends 'COMMANDS' ? ControlCommand : Annotation;
+}
 export class InferenceResponse {
-  constructor(
-    public type: string,
-    public data: ControlCommand | Annotation,
-  ) {}
-
-  static fromJson(json: unknown, resizeRatio = 1, image?: string):
-  ControlCommand | Annotation {
-    const inferenceResponse = json as InferenceResponse;
-    return this.createModels(inferenceResponse.type, inferenceResponse.data, resizeRatio, image);
+  static fromJson(
+    json: InferenceResponseBody,
+    resizeRatio = 1,
+    image?: string,
+  ): ControlCommand | Annotation {
+    return this.createModels(json.type, json.data, resizeRatio, image);
   }
 
   static createModels(
-    type: string,
-    data: ControlCommand | Annotation,
+    type: ModelType,
+    data: ModelType extends 'COMMANDS' ? ControlCommand : Annotation,
     resizeRatio: number,
     image?: string,
-  ):
-    ControlCommand | Annotation {
-    return this.models[type as keyof Models](data, resizeRatio, image);
+  ): ControlCommand | Annotation {
+    if (type === 'COMMANDS') return ControlCommand.fromJson(data, resizeRatio);
+    if (type === 'DETECTED_ELEMENTS') {
+      return Annotation.fromJson(
+        { image, detected_elements: data.detected_elements },
+        resizeRatio,
+      );
+    }
+    throw new InvalidModelTypeError(type);
   }
-
-  static models: Models = {
-    DETECTED_ELEMENTS: (
-      data: Annotation,
-      resizeRatio: number,
-      image: string,
-    ) => Annotation
-      .fromJson({ image, detected_elements: data.detected_elements }, resizeRatio),
-    COMMANDS: (data: ControlCommand, resizeRatio: number) => ControlCommand
-      .fromJson(data, resizeRatio),
-  };
-}
-
-interface Models {
-  DETECTED_ELEMENTS: CallableFunction
```

Comment:
can you explain why you changed it?

Reply:
Typing was incorrect in some places. Fixing that lead to the typing here being incorrect etc. so I changed it as well...

Reply:
my question was mainly regarding introducing the if conditions.

Reply:
As easy to change and read as the map option so not degregation in code quality  but easier typing...